### ISSUE TITLE
Shift+Tab to radio button choice group focuses on last option but does not select

### ISSUE THREAD

**Issue by dylankil** [OfficeDev/office-ui-fabric-react#3667]
### Bug Report

Package version: current
Browser and OS versions: All

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix?
Yes, but I need a better understanding of the project.

Requested priority: Normal

Products/sites affected: All

While tabbing through content Shift+Tab from the element immediately after a radio button choice group, note the last radio button choice group option is focused but not selected

#### Actual behavior:

The last radio button choice group option is focused but not selected

#### Expected behavior:

The last radio button choice group option is focused but not selected, but the user should be able to use the enter key to select the currently focused option

**Comment by natalieethell** [comment]
Hey Dylan, thanks so much for submitting this issue! A question for you. So right now, the ChoiceGroup seems to have the following behavior:

1. If we already have an option selected.
I can tab (or shift + tab) through the page, which lands me on the selected option. From there, I can use the arrow keys to shift focus and select an option at the same time.

2. If no option is selected yet.
I can tab through the page, which focuses on the first option OR I can shift + tab through the page, which focuses on the last option. From there, I can use the arrow keys to shift focus and select an option at the same time.

Correct me if I'm wrong, but would you prefer being able to hit enter over using the arrow keys to shift focus and select? I'm going to tag @betrue-final-final on this, who may have a bit to say on this change and determining whether focus and selection should be one action or two separate ones.

**Comment by dylankil** [comment]
We’d prefer that if an unselected option is focused the enter key should select it, using the arrow keys in this scenario adds unnecessary clicks for People With Disabilities.

**Comment by natalieethell** [comment]
Hey Dylan, I've been looking into this issue a little more.

It looks like most other places use arrow keys to change focus and selection on options within the radio button group: [w3.org](https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/radio/radio-1/radio-1.html), chrome://settings/

However, hitting enter to select the currently focused (but unselected) option in a radio button group seems to only be supported by about half of the places in which I've seen radio buttons. The [w3.org](https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/radio/radio-1/radio-1.html) example supports this. A plain input of type "radio" does not support this. Here's a CodePen example: https://codepen.io/anon/pen/qKEpjo.

@betrue-final-final would we like to maintain the logic of using arrow keys to change selection on options within the radio button group and change our logic to support hitting enter (or space) to select a currently focused, but unselected option?

**Comment by betrue-final-final** [comment]
Arrow should only move focus, and space and optionally enter should select/deselect. 

Otherwise it will be difficult to control selection. 

This is how the control has worked for decades.

**Comment by natalieethell** [comment]
Hm, it looks like everywhere I've seen has arrows moving selection. I cannot seem to find an example where arrows only move focus. Do you have a good example?

**Comment by dylankil** [comment]
Sorry for the misunderstanding. The bug that was filed by our test team is specifically when shift tabbing from a different component. Arrow keys should shift focus and select, however, the user should be able to use the enter or space key to select the last option as this is the option focused when shift tabbing from a focusable element that appears after the ChoiceGroup.

Is there any further clarification I can provide?

**Comment by natalieethell** [comment]
Hey @dylankil no worries! I think I understand what you're looking for, thank you for the clarification. :)
 
@betrue-final-final Here's an example of what I'm seeing on w3.org and Chrome, where arrow keys do change selection, not just focus.
![radio buttons w3](https://user-images.githubusercontent.com/14134000/40744713-4a8f0438-640a-11e8-8fd6-0825aa4acea9.gif)
![radio buttons chrome](https://user-images.githubusercontent.com/14134000/40744721-4e4e05a6-640a-11e8-90b0-66726563bbee.gif)

I can't seem to find an example where this isn't the case. Do you know of somewhere that has the behavior you're looking for?

**Comment by betrue-final-final** [comment]
It appears that the web standard is different than I recall for Win32. There is a subtlety in tabbing into the control does not select the control, until the arrow key is used. https://www.w3.org/wiki/RadioButton 
Space does seem to toggle selection.
Focus can move to a radio group via:

- The Tab key
- An Access Key or mnemonic targeting the label
- Activation of the label (through Assistive Technology mechanism)
- The Tab key moves focus between radio button groups and other widgets.
- When focus is on the group and when no radio button is selected:
- Tab key press moves focus to the first radio button in the group, but does not select the radio button.
- Shift+Tab key press moves focus to the last radio button in the group, but does not select the radio button.
- When focus moves to the group in which a radio button is selected, pressing Tab and Shift+Tab keys move focus to the radio button that is checked.
- Up Arrow and Down Arrow keys move focus and selection.
- Up Arrow key press moves focus and selection forward through the radio buttons in the group. If the first radio button has focus, then focus and selection move to the last radio button in the group.
- Down Arrow key press moves focus and selection backward through the radio buttons in the group. If the last radio button has focus, then focus and selection move to the first radio button in the group. * Space Bar key press checks the radio button that currently has focus.
- When re-entering the group, focus returns to the point of previous focus (the item with selection set).
- Pressing the Tab key exits the group and moves focus to the next form control.
- Pressing the Shift+Tab keys exits the group and moves focus to the previous form control.

**Comment by natalieethell** [comment]
@betrue-final-final To me, it sounds like it might be good to stick to the W3C standard on this one and maintain the current behavior. What do you think?

**Comment by natalieethell** [comment]
@dylankil It looks like we will maintain the current behavior of the radio buttons, which aligns with the W3C standard. I will close this issue.
